opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -296,277 +196,151 @@ class Restart : private boost::noncopyable
-    }
-
-    // no destructor in this class as ERT takes care of freeing the
-    // handle as part of freeing the solution handle!
-
-    ecl_sum_tstep_type *ertHandle() const
-    { return ertHandle_; };
-
-private:
-    ecl_sum_tstep_type *ertHandle_;
-};
-
-
 /**
  * Initialization file which contains static properties (such as
  * porosity and permeability) for the simulation field.
  */
-class Init : private boost::noncopyable
-{
+class Init {
 public:
-    Init(const std::string& outputDir,
-         const std::string& baseName,
-         int writeStepIdx,
-         IOConfigConstPtr ioConfig) : egridFileName_(outputDir,
-                                                     baseName,
-                                                     ECL_EGRID_FILE,
-                                                     writeStepIdx,
-                                                     ioConfig->getFMTOUT())
-    {
-        bool formatted = ioConfig->getFMTOUT();
-
-        FileName initFileName(outputDir,
-                              baseName,
-                              ECL_INIT_FILE,
-                              writeStepIdx,
-                              formatted);
-
-        ertHandle_ = fortio_open_writer(initFileName.ertHandle(),
-                                        formatted,
-                                        ECL_ENDIAN_FLIP);
-    }
+    Init( const std::string& outputDir,
+          const std::string& baseName,
+          int writeStepIdx,
+          const IOConfig& ioConfig ) :
+        egrid( outputDir,
+               baseName,
+               ECL_EGRID_FILE,
+               writeStepIdx,
+               ioConfig.getFMTOUT() ),
+        init( fortio_open_writer(
+                    FileName( outputDir,
+                        baseName,
+                        ECL_INIT_FILE,
+                        writeStepIdx,
+                        ioConfig.getFMTOUT() ).ertHandle(),
+                    ioConfig.getFMTOUT(),
+                    ECL_ENDIAN_FLIP ) )
 
-    ~Init()
-    { fortio_fclose(ertHandle_); }
+    {}
 
-    void writeHeader(int numCells,
-                     const int* compressedToCartesianCellIdx,
-                     const SimulatorTimerInterface& timer,
-                     Opm::EclipseStateConstPtr eclipseState,
-                     const PhaseUsage uses,
-                     const NNC& nnc = NNC())
+    void writeHeader( int numCells,
+                      const int* compressedToCartesianCellIdx,
+                      time_t current_posix_time,
+                      const EclipseState& es,
+                      int ert_phase_mask,
+                      const NNC& nnc = NNC() )
     {
-        auto dataField = eclipseState->get3DProperties().getDoubleGridProperty("PORO").getData();
-        restrictAndReorderToActiveCells(dataField, numCells, compressedToCartesianCellIdx);
+        auto dataField = es.get3DProperties().getDoubleGridProperty("PORO").getData();
```

Comment:
I thought that the intention of the ECL restart files is to output what the simulator uses and not repeat what the deck says. Given that this is ECL, I'm not too much surprised if that is not the case, though. (I'm just wondering what the value proposition of that data is then.)

> You are right that the simulator uses PORV - which is either initialized from cell volumes, POROand NTG or entered explicitly in the grid. We should output PORV - from the Eclipse3DProperties array[1]; and if present in the grid we should output PORO and also NTG. But the fact the simulator uses PORV does not in any way warrant the claim that we should not output PORO.

do you mean that an additional check for PORO is needed because it is not present in some decks? In that case, I agree, but what happens for "partially defaulted" PORO fields where the defaulted cells are specified via PORV?